test/sql/cast/struct_to_map.test

```diff
@@ -0,0 +1,328 @@
+# name: test/sql/cast/struct_to_map.test
+# group: [cast]
+
+# enable query verification
+statement ok
+PRAGMA enable_verification
+
+# Basic functionality tests
+
+# Simple cast - VARCHAR keys to VARCHAR values
+query I
+SELECT {"name": 'Alice', "city": 'NYC'}::MAP(VARCHAR, VARCHAR);
+----
+{name=Alice, city=NYC}
+
+# Simple cast - VARCHAR keys to INT values
+query I
+SELECT {"age": 25, "score": 100}::MAP(VARCHAR, INT);
+----
+{age=25, score=100}
+
+# Multiple rows with same field structure
+query I
+SELECT col::MAP(VARCHAR, INT)
+FROM VALUES ({"duck": 1}),({"duck": 2}),({"duck": 3}),({"duck": 4}),({"duck": 5})
+AS tab(col);
+----
+{duck=1}
+{duck=2}
+{duck=3}
+{duck=4}
+{duck=5}
+
+# Multiple fields in a single struct
+query I
+SELECT {"name": 'Alice', "age": 25, "score": 95}::MAP(VARCHAR, VARCHAR);
+----
+{name=Alice, age=25, score=95}
+
+# Key type casting tests
+
+# Cast VARCHAR keys to INT keys (keeps null values)
+query I
+SELECT col::MAP(INT, INT)
+FROM VALUES ({"1": 1}),({"2": 2}),({"3": 3}),({"4": 4}),({"5": 5})
```

Comment:
Im not sure these become STRUCT, because we have multiple rows with different shaped structs

Im actually confused this worked if I'm honest

Reply:
Ah actually, if you look at the result, we combine the structs, into `STRUCT('1' INTEGER, '2' INTEGER, ...)` so you can disregard this comment